File: tests/integration/construct_component_plain/dotnet/Component.cs
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

using Pulumi;

class Component : Pulumi.ComponentResource
{
    public Component(string name, ComponentArgs? args = null, ComponentResourceOptions? opts = null)
        : base("testcomponent:index:Component", name, args ?? ResourceArgs.Empty, opts, remote: true)
    {
    }
}

public sealed class ComponentArgs : Pulumi.ResourceArgs
{
    [Input("children")]
    public int? Children { get; set; }

    public ComponentArgs()
    {
    }
}

File: tests/integration/construct_component_plain/nodejs/component.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

export interface ComponentArgs {
    children?: number;
}

export class Component extends pulumi.ComponentResource {
    constructor(name: string, args: ComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("testcomponent:index:Component", name, args, opts, true);
    }
}
